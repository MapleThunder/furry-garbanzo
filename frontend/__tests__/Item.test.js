import ItemComponent from "../components/Item";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

const fakeItem = {
  id: 123,
  title: "💎 💎 🙌🏻 🙌🏻",
  price: 69420,
  description: "Squeeze tight",
  image: "diamond.jpg",
  largeImage: "diamond_lg.jpg",
};

describe("<Item />", () => {
  it("renders and matches the snapshot", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  // it("renders the image properly", () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);

  //   const img = wrapper.find("img");
  //   expect(img.props().src).toBe(fakeItem.image);
  //   expect(img.props().alt).toBe(fakeItem.title);
  // });

  // it("renders the price tag and title properly", () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const PriceTag = wrapper.find("PriceTag");

  //   expect(PriceTag.children().text()).toBe("$4.20");
  //   expect(wrapper.find("Title a").text()).toBe(fakeItem.title);
  // });

  // it("renders the buttons properly", () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const buttonList = wrapper.find(".buttonList");

  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find("Link")).toHaveLength(1);
  //   expect(buttonList.find("AddToCart").exists()).toBe(true);
  //   expect(buttonList.find("DeleteItem").exists()).toBe(true);
  // });
});
