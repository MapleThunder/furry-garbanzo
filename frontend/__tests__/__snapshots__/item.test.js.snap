// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="💎 💎 🙌🏻 🙌🏻"
    src="diamond.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": 123,
          },
        }
      }
    >
      <a>
        💎 💎 🙌🏻 🙌🏻
      </a>
    </Link>
  </Title>
  <PriceTag>
    $694.20
  </PriceTag>
  <p>
    Squeeze tight
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": 123,
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id={123}
    />
    <DeleteItem
      id={123}
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
